File: jscomp/test/gpr_2682_test.ml

(* [@@@bs.config no_export] *)
(* let for_each n =
  ([%raw{|
	for (var key in n){
      console.log(key)
    }
  |}] : unit ); () *)



let sum : int -> int -> int  = fun%raw a b -> {| 
  return a + b
|}

let v = sum 1 2 

let f a = a + [%raw {|3|}] 


let b = f 1 
let c = f 2 

let forIn = 
   (fun%raw  o foo -> {|
  for (var i in o){
    foo(o)
  }
  |})


let forIn : 'a -> (string -> unit [@bs]) -> unit = forIn  


(*
let%raw forIn : 'a -> (string -> unit [@bs]) -> unit = fun o foo -> {|
  for (var i in o){
    foo(o)
  }
|}
*)
module N  : sig 
  val log2 : string -> unit [@bs]
end  = struct 
let log = fun [@bs] x -> Js.log x 

let log2 : 'a -> unit [@bs] = log 
end 

(* let log : 'a -> unit = fun%raw x -> {|console.log (x)|}   *)

;;   forIn   [%obj{x = 3 }]  (fun[@bs] x -> Js.log x)
;;   forIn   [%obj{x = 3 ; y = 3}]  (fun[@bs] x -> Js.log x)


let f3  : unit -> bool [@bs] = fun%raw () -> "return true"

let bbbb  = f3 () [@bs]

File: jscomp/test/gpr_2682_test.js
'use strict';


var sum = ((a,b) => a + b);

var v = sum(1, 2);

function f(a) {
  return a + 3 | 0;
}

var b = f(1);

var c = f(2);

var forIn = ((o,foo)=> {
  for (var i in o){
    foo(o)
  }
  });

function log(x) {
  console.log(x);
  
}

var N = {
  log2: log
};

forIn({
      x: 3
    }, (function (x) {
        console.log(x);
        
      }));

forIn({
      x: 3,
      y: 3
    }, (function (x) {
        console.log(x);
        
      }));

var f3 = (()=>true);

var bbbb = f3();

if (!bbbb) {
  throw {
        RE_EXN_ID: "Assert_failure",
        _1: [
          "gpr_2682_test.ml",
          59,
          2
        ],
        Error: new Error()
      };
}

exports.sum = sum;
exports.v = v;
exports.f = f;
exports.b = b;
exports.c = c;
exports.forIn = forIn;
exports.N = N;
exports.f3 = f3;
exports.bbbb = bbbb;
/* v Not a pure module */
